### ISSUE TITLE
Segmentation faults when running flow_cp with norne_default.param

### ISSUE THREAD

**Issue by blattms** [OPM/opm-autodiff#457]
I am getting some hard to debug segementation faults running norne with flow_cp (flow works) compiled with "-O2 -DNDEBUG":
```
...
--- Scaling parameter 'ISGCR' assigned.
--- Scaling parameter 'ISOWCR' assigned.
--- Scaling parameter 'ISOGCR' assigned.
Can only handle a single region in ROCK (2 regions specified). Ignoring all except for the first.
gravity not found. Using default value '9.80665'.

Program received signal SIGSEGV, Segmentation fault.
0x00000000006138ce in std::vector<std::vector<double, std::allocator<double> >, std::allocator<std::vector<double, std::allocator<double> > > > Opm::Equil::phasePressures<Dune::CpGrid, Opm::Equil::EquilReg<Opm::Equil::DensityCalculator<Opm::BlackoilPropertiesInterface> >, Opm::RegionMapping<std::vector<int, std::allocator<int> > >::CellRange>(Dune::CpGrid const&, Opm::Equil::EquilReg<Opm::Equil::DensityCalculator<Opm::BlackoilPropertiesInterface> > const&, Opm::RegionMapping<std::vector<int, std::allocator<int> > >::CellRange const&, double) ()

```

BTW this is happening on two of my systems: Debian wheezy (gcc-4.7.2) and jessie (gcc-4.9.2) no matter whether I use MPI/Zoltan or not. SPE cases work fine.

As I do not get the problem with non-optimized debugging builds I will do some printf debugging in ``OPM::equil::phasePressure```.
Any hints for suspects are highly welcome.

**Comment by andlaus** [comment]
this looks a bit like an ABI issue due to the binaries of different libraries expecting different object sizes. Since you cannot be using the latest master version of opm-core here, would you mind finding out if this issue goes away if you update every module to the current master version and do a "full" recompile of everything?

**Comment by blattms** [comment]
I did the build on the wheezy machine this morning with freshly clone repositories (except for ert).

**Comment by andlaus** [comment]
what you say cannot be true because the latest master version does not print messages like

```
--- Scaling parameter 'ISOWCR' assigned.
```

anymore. are you _really_ sure that there was no mistake on the way?

**Comment by osae** [comment]
Note that a non-zero "low" in RegionMapping.hpp might cause this kind of behavior, i.e cellrange outside valid memory.  (This might perhaps happen for norne if the equilibration works on sub-partitions of the grid.)

**Comment by bska** [comment]
> Note that a non-zero "low" in `RegionMapping.hpp` might cause this kind of behavior

I'm working on that.

**Comment by blattms** [comment]
@andlaus maybe the latest doesn't, but the one from around noon did. I did not have any dune/opm on that machine.
@osae printf said that the first cell was 0. That should be ok.

Anyway, I seem to have found the issue and can be named the proud author of this Heisenbug :(. Just having a hard time understanding what really is wrong. Will post a PR.

**Comment by andlaus** [comment]
> @andlaus maybe the latest doesn't, but the one from around noon did. I did not have any dune/opm on that machine.

atgeirr merged OPM/opm-core#834 yesterday, didn't he?

**Comment by blattms** [comment]
@andlaus We are both right: My posted output is from my jessie machine with the older version, but I verified it on a fresh checkout on the other machine. Should have the output from there. Sorry for the confusion.

It is not an ABI problem, but one of const refernces to temporary objects not being handled like they should.
